vendor/parser-extended/ast_mapper.ml

```diff
@@ -316,7 +316,7 @@ module CT = struct
 
   let map_signature sub {pcsig_self; pcsig_fields} =
     Csig.mk
-      (sub.typ sub pcsig_self)
+      ({pcsig_self with pcss_desc = map_opt (sub.typ sub) pcsig_self.pcss_desc})
```

Comment:
Personally I like having a function for each type (eg. `map_class_sig_self`), and if we keep the loc (see my comment on parsetree, I don't think we should), the loc needs to be mapped as well with `sub.location`.